### PR TITLE
Get parsed matches

### LINKED ISSUE #1907: store parsed matches in ParseHistory table
A common request is to get IDs of only matches that are parsed. Currently we don't index those, but we can create a new table and store the IDs there.

### PR DESCRIPTION
**Pull request by ahi-el**
fixes #1907

added a url path to get parsed matches, limited by 100 & using less_than_match_id

### CODE REVIEW

File: routes/spec.js [hunk 1]
```diff
@@ -2325,6 +2325,49 @@ You can find data that can be used to convert hero and ability IDs and other inf
...
+            },
+          },
+        },
+        route: () => '/parsedMatches',
+        func: (req, res, cb) => {
+          const lessThan = Number(req.query.less_than_match_id) || Number.MAX_SAFE_INTEGER;
+
+          db.raw(`
+          SELECT * FROM parsed_matches
+          WHERE match_id < ?
```
Comment (Argn0) [inline routes/spec.js]: in my opinion less than or equal would be  more intuitive

File: routes/spec.js [hunk 2]
```diff
@@ -2325,6 +2325,49 @@ You can find data that can be used to convert hero and ability IDs and other inf
...
+        responses: {
+          200: {
+            description: 'Success',
+            schema: {
+              type: 'array',
+              items: {
+                type: 'object',
+                properties: {
+                  match_id: {
+                    description: 'match_id',
```
Comment (Argn0) [inline routes/spec.js]: I find this description too inaccurate

File: routes/spec.js (hunk 1 above)
Comment (Argn0) [reply routes/spec.js]: probably doesn't matter though

File: routes/spec.js (hunk 2 above)
Comment (ahi-el) [reply routes/spec.js]: What would you say is inaccurate about the description? match_id is pretty self explanatory.

Should it be something like "match id of a previously parsed match"?

File: routes/spec.js (hunk 1 above)
Comment (ahi-el) [reply routes/spec.js]: sounds good

File: routes/spec.js (hunk 2 above)
Comment (Argn0) [reply routes/spec.js]: nevermind. I have mistaken this for the query paremeter.

File: routes/spec.js [hunk 3]
```diff
@@ -2325,6 +2325,50 @@ You can find data that can be used to convert hero and ability IDs and other inf
         },
       },
     },
+    '/parsedMatches': {
+      get: {
+        summary: 'GET /parsedMatches',
+        description: 'Get list of parsed matches',
```
Comment (howardchung) [inline routes/spec.js]: I'd change this to "match IDs"

File: routes/spec.js [hunk 4]
```diff
@@ -2325,6 +2325,50 @@ You can find data that can be used to convert hero and ability IDs and other inf
...
+        tags: ['parsed public matches'],
```
Comment (howardchung) [inline routes/spec.js]: "parsed matches", this will include all

File: routes/spec.js [hunk 5]
```diff
@@ -2325,6 +2325,50 @@ You can find data that can be used to convert hero and ability IDs and other inf
...
+                    type: 'integer',
+                  },
+                },
+              },
+            },
+          },
+        },
+        route: () => '/parsedMatches',
+        func: (req, res, cb) => {
+          const lessThan = Number(req.query.less_than_match_id) || Number.MAX_SAFE_INTEGER;
```
Comment (howardchung) [inline routes/spec.js]: I am pretty sure you can just pass this as a string to postgres and let the database handle it.

File: routes/spec.js [hunk 6]
```diff
@@ -2325,6 +2325,50 @@ You can find data that can be used to convert hero and ability IDs and other inf
...
+
+          db.raw(`
+          SELECT * FROM parsed_matches
+          WHERE match_id <= ?
```
Comment (howardchung) [inline routes/spec.js]: should be <

File: routes/spec.js (hunk 1 above)
Comment (howardchung) [reply routes/spec.js]: I think we should use < for consistency with the other method and the name

File: routes/spec.js (hunk 6 above)
Comment (ahi-el) [reply routes/spec.js]: done

File: routes/spec.js (hunk 4 above)
Comment (ahi-el) [reply routes/spec.js]: done

File: routes/spec.js (hunk 3 above)
Comment (ahi-el) [reply routes/spec.js]: done

File: routes/spec.js (hunk 5 above)
Comment (ahi-el) [reply routes/spec.js]: I'm not sure what you mean by this. Can you give an example?

Comment (howardchung) [reply routes/spec.js]: you can just remove the Number() cast

Comment (ahi-el) [reply routes/spec.js]: ah, got it now. Done

**Pull request merged**